filer/templatetags/filer_admin_tags.py

```diff
@@ -34,3 +34,23 @@ def filer_admin_context_hidden_formfields(context):
         '<input type="hidden" name="{0}" value="{1}">',
         admin_url_params(request).items(),
     )
+
+
+@register.assignment_tag(takes_context=True)
+def filer_has_permission(context, item, action):
+    """Does the current user (taken from the request in the context) have
+    permission to do the given action on the given item.
+
+    """
+    permission_method_name = 'has_{action}_permission'.format(action=action)
+    permission_method = getattr(item, permission_method_name, None)
+    request = context.get('request')
+
+    if not permission_method or not request:
+        return False
+    # Try to call the permission method.
+    # This amounts to calling `item.has_X_permission(request)`
+    try:
```

Comment:
the original implementation does not do this, please avoid such wide error handling.
If this method fails then it should fail loudly.